src/App.jsx

```diff
@@ -14,33 +15,83 @@
   'Garlic',
 ];
 
-export const App = () => (
+const sortFieldAlphabet = 'alphabet';
+const sortFieldLength = 'length';
+const reverseFieldGoods = 'reverse';
+
+export const App = () => {
+  const [visibleGoods, setVisibleGoods] = useState(goodsFromServer);
+  const [activeSort, setActiveSort] = useState('');
+  const [originalGoods, setOriginalGoods] = useState(goodsFromServer);
+
+  const resetGoodsOfOrder = () => {
+    setVisibleGoods(goodsFromServer);
+    setActiveSort('');
+  };
+
+  const sortGoodsAlphabetically = () => {
+    setVisibleGoods(
+      [...visibleGoods].sort((good1, good2) => good1.localeCompare(good2))
+    );
+    setActiveSort(sortFieldAlphabet);
+  };
+
+  const sortGoodsByLength = () => {
+    setVisibleGoods(
+      [...visibleGoods].sort((good1, good2) => good1.length - good2.length)
+    );
+    setActiveSort(sortFieldLength);
+  }
+
+  const reverseGoodsOrder = () => {
+    setVisibleGoods(
+      [...visibleGoods].reverse()
+    );
+    setActiveSort(reverseFieldGoods)
+  }
+
+  return (
   <div className="section content">
     <div className="buttons">
-      <button type="button" className="button is-info is-light">
+      <button
+        onClick={sortGoodsAlphabetically}
+        type="button"
+        className={`button is-info ${activeSort === sortFieldAlphabet ? '' : 'is-light'}`}
+      >
         Sort alphabetically
       </button>
 
-      <button type="button" className="button is-success is-light">
+      <button
+        onClick={sortGoodsByLength}
+        type="button"
+        className={`button is-success ${activeSort === sortFieldLength ? '' : 'is-light'}`}
+      >
         Sort by length
       </button>
 
-      <button type="button" className="button is-warning is-light">
+      <button
+        onClick={reverseGoodsOrder}
+        type="button"
+        className={`button is-warning ${activeSort === reverseFieldGoods ? '' : 'is-light'}`}
+      >
         Reverse
       </button>
-
-      <button type="button" className="button is-danger is-light">
+     {JSON.stringify(visibleGoods) !== JSON.stringify(originalGoods) && (
```

Comment:
In line 79, you are using `JSON.stringify()` to compare arrays. This can be inefficient for large arrays as it needs to convert the whole array into a string. Consider using a more efficient way to compare arrays, for example using `_.isEqual()` from lodash or writing a custom function. Here is a simple example of a custom function that compares arrays:  function arraysAreEqual(array1, array2) { if (array1.length !== array2.length) return false; for (let i = 0; i < array1.length; i++) { if (array1[i] !== array2[i]) return false; } return true; } ``` You can use it like this: `arraysAreEqual(visibleGoods, originalGoods)`. However, please note that this function assumes that the order of the elements in the arrays matters. If the order doesn't matter, you would need a more complex function.